src/method.c

```diff
@@ -149,6 +152,28 @@ static jl_value_t *resolve_globals(jl_value_t *expr, jl_module_t *module, jl_sve
             if (e->head == method_sym || e->head == module_sym) {
                 i++;
             }
+            if (e->head == new_opaque_closure_sym) {
+                JL_NARGSV(opaque closure definition, 5);
+                for (; i < 4; i++) {
+                    // TODO: this should be making a copy, not mutating the source
+                    jl_exprargset(e, i, resolve_globals(jl_exprarg(e, i), module, sparam_vals, binding_effects, eager_resolve));
+                }
+                jl_value_t *source = jl_exprarg(e, 4);
+                if (jl_is_expr(source) && strcmp(jl_symbol_name(((jl_expr_t*)source)->head), "opaque_closure_method") == 0) {
```

Comment:
It looks to me like now only `opaque_closure_method` needs to be a syntax form, and `new_opaque_closure` is effectively a function. IIUC, the only reason `opaque_closure_method` needs to be a special form is to call jl_make_opaque_closure_method when the enclosing top-level expression is evaluated?

Reply:
`new_opaque_closure` has magic semantics once inference is enabled (not part of this PR), which is why I was asked to make it an IR form rather than a function call.